Code/BuildSystem/CMake/CMakeUtils/ezUtilsUnityFiles.cmake

```diff
@@ -63,26 +65,30 @@ function(ez_generate_folder_unity_file TARGET_NAME PROJECT_DIRECTORY SUB_FOLDER_
 		set(CUR_FILE_FULL "${PROJECT_DIRECTORY}/${SUB_FOLDER_PATH}/${CUR_FILE}")
 
 		file(TO_NATIVE_PATH ${CUR_FILE_FULL} CUR_FILE_NATIVE)
-		file(APPEND ${TMP_UNITY_FILE} "#include \"${CUR_FILE_NATIVE}\"\n")
+		string(APPEND UNITY_FILE_CONTENTS "#include \"${CUR_FILE_NATIVE}\"\n")
 
 		set_source_files_properties(${CUR_FILE_FULL} PROPERTIES HEADER_FILE_ONLY true)
 	endforeach()
+	
```

Comment:
Shouldn't this write the file, rather than append?

Reply:
No. The idea behind this optimization is that this cmake scripts constructs a string that contains the entire new content of the unity cpp file. Then if the sha1 check shows that a modification happend, the file is written to disk. This reduces and amount of file operations cmake does, and speeds up the unity cpp generation.

Reply:
Yes, I get that part. But at least on my phone I didn't see that you ever write the file, ie replace it, rather you seem to only append, which may work if the file doesn't exist, but not if it does. So that was my question.